src/neuralnet/cpid.cpp

```diff
@@ -82,6 +82,16 @@ bool Cpid::operator!=(const Cpid& other) const
     return m_bytes != other.m_bytes;
 }
 
+bool Cpid::operator<(const Cpid& other) const
+{
+    return m_bytes < other.m_bytes;
+}
+
+bool Cpid::operator>(const Cpid& other) const
+{
+    return m_bytes > other.m_bytes;
```

Comment:
Out of curiosity, what is standard when writing comparison operators? I usually write one and use it negated in the other.

Reply:
Does it matter? Isn’t negating two operations versus a comparison one. Also writing for each operator is clearer.

Reply:
We technically don’t need one of those unless we want to sort the maps in the opposite order, but I prefer to be complete.

Reply:
My rationale was that you'd only have to write and maintain one. The others comes for free as "not the first one". If that makes sense :) Maybe it makes more sense when you have more going on inside the comparators.

Reply:
I think it makes a lot of sense if the comparison operators are non-trivial. Here they are trivial, so I will leave it unless you really object...

Reply:
I read advice about negating when I was learning about operator overloading, and I do that in other places. It makes sense, but I thought it was easier to read the plain comparison for the trivial ones. Happy to change it, though.

Reply:
It is fine by me. I was just wondering if there was a standard way to pick a path. Since these are trivial I agree with keeping them as is.